src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc

```diff
@@ -1,23 +1,251 @@
 [id="devfile-reference_{context}"]
 = Devfile reference
 
-This paragraph is the reference module introduction and is only optional. Include it to provide a short overview of the module.
+This section contains devfile reference and instructions on how to use the various elements that devfiles consist of.
 
-A reference module provide data that users might want to look up, but do not need to remember. It has a very strict structure, often in the form of a list or a table. A well-organized reference module enables users to scan it quickly to find the details they want. AsciiDoc markup to consider for reference data:
+== Adding components to a devfile
 
+Each component in a single devfile must have a unique name.
 
-.Unordered lists
-* For more details on writing reference modules, see the link:https://github.com/redhat-documentation/modular-docs#modular-documentation-reference-guide[Modular Documentation Reference Guide].
-* Use a consistent system for file names, IDs, and titles. For tips, see _Anchor Names and File Names_ in link:https://github.com/redhat-documentation/modular-docs#modular-documentation-reference-guide[Modular Documentation Reference Guide].
+=== Component type: cheEditor
 
-.Labeled lists
-Term 1:: Definition
-Term 2:: Definition
+Describes the editor used in the workspace by defining its `id`. A devfile can only contain one component of the `cheEditor` type.
 
-.Tables
-[options="header"]
-|====
-|Column 1|Column 2|Column 3
-|Row 1, column 1|Row 1, column 2|Row 1, column 3
-|Row 2, column 1|Row 2, column 2|Row 2, column 3
-|====
+[source,yaml]
+----
+components:
+  - alias: theia-editor
+    type: cheEditor
+    id: eclipse/che-theia/next
+----
+
+When `cheEditor` is missing, a default editor is provided along with its default plug-ins. The default plug-ins are also provided for an explicitly defined editor with the same `id` as the default one (even if it is a different version). Che-Theia is configured as default editor along with the *Che Machine Exec* plug-in.
+
+To specify that a workspace requires no editor, use the `editorFree:true` attribute in the devfile attributes.
```

Comment:
Should we put an example (a reference to example) here?

Reply:
I added a link to the section that describes the attribute.